TFNetworkLayer.py

```diff
@@ -2867,14 +2868,24 @@ def __init__(self, axis=None, energy_factor=None,
           tf.ones_like(window_start) * (energy_shape[axis] - window_len),
           window_start)  # case 2.
         n_batch = energy_shape[energy_data.batch_dim_axis]
-        indices = expand_dims_unbroadcast(tf.range(window_len), energy_data.batch_dim_axis, n_batch)
-        # time-major: (W,B) + (1,B), batch-major: (B, W) + (1,B)
-        indices += tf.expand_dims(tf.to_int32(window_start), axis=axis)
-        idxs = nd_indices(indices)
-        mask_shape = energy_shape[:2]  # (T, B)
-        mask_shape[axis] = window_len
-        energy_mask_window = tf.scatter_nd(idxs, tf.ones(shape=mask_shape), energy_shape[:2])
+        indices = expand_dims_unbroadcast(tf.range(window_len), 0, n_batch)  # (B, W)
+        # (B, W) + (B, 1)
+        indices += tf.expand_dims(tf.to_int32(window_start), axis=1)  # + (B, 1)
+        indices.set_shape((None, None))  # required for older TF versions
+        idxs = nd_indices(indices)  # (B, W, 2)
+        mask_shape = [n_batch, window_len]  # (B, W)
+        rem_energy_data = energy_data.copy_with_time_dim_axis(time_dim_axis=1)  # (B, T, ...)
+        rem_energy_shape = tf.shape(rem_energy_data.placeholder, name="rem_energy_shape")
+        rem_energy_shape = [rem_energy_shape[i] for i in range(rem_energy_data.batch_ndim)]
+        rem_axes = list(energy_shape)
+        rem_axes.pop(energy_data.batch_dim_axis)
+        rem_axes.pop()  # (B, ..., T) -> (...)
```

Comment:
`pop` removes by default the first or last? Maybe better make that explicit.
Is the comment correct? Above you do `rem_energy_data. copy_with_time_dim_axis(1)`, so time is at axis 1, not at the end?
Also, does this code assume that batch dim axis is 0 or 1? What if it is bigger?

Reply:
`pop` removes the last item by default. I made it more explicit.
`rem_axes.pop()` operates on a copy of `energy_shape`, and not `rem_energy_data`.
I've moved the code a bit further down, so that it is not confusing at first glance.

I haven't tested it with different batch-dim axes, but it does convert it into batch-major.